state-chain/runtime/src/lib.rs

```diff
@@ -2072,8 +2072,41 @@ impl_runtime_apis! {
 		fn cf_validate_refund_params(retry_duration: u32) -> Result<(), DispatchErrorWithMessage> {
 			pallet_cf_swapping::Pallet::<Runtime>::validate_refund_params(retry_duration).map_err(Into::into)
 		}
+
+		fn cf_open_btc_deposit_channels(account_id: AccountId) -> Result<Vec<<cf_chains::Bitcoin as cf_chains::Chain>::ChainAccount>, DispatchErrorWithMessage> {
+			Ok(pallet_cf_ingress_egress::DepositChannelLookup::<Runtime,BitcoinInstance>::iter()
+				.map(|(_, value)| value)
+				.filter(|channel_details| channel_details.owner == account_id)
+				.filter(|channel_details| match channel_details.action {
+					ChannelAction::Swap {..} => true,
+					ChannelAction::CcmTransfer {..} => true,
+					ChannelAction::LiquidityProvision {..} => false,
+				})
+				.map(|channel_details| channel_details.deposit_channel.address)
```

Comment:
I don't really mind, but clippy will probably complain and tell you merge this filter.map.map sequence into a single filter_map.

Reply:
Clippy is happy as is, and so am I :)

Reply:
I'm glad, in these cases I think I prefer filter().map(), reads more nicely than shoving an if statement in a filter_map...